core/vtk/ttkMergeTreeAutoencoder/ttkMergeTreeAutoencoder.h

```diff
@@ -0,0 +1,591 @@
+/// \ingroup vtk
+/// \class ttkMergeTreeAutoencoder
+/// \author Mathieu Pont <mathieu.pont@lip6.fr>
+/// \date 2023.
+///
+/// \brief TTK VTK-filter that wraps the ttk::MergeTreeAutoencoder module.
+///
+/// This VTK filter uses the ttk::MergeTreeAutoencoder module to compute an
+/// auto-encoder of merge trees or persistence diagrams.
+///
+/// \param Input vtkMultiBlockDataSet Input trees
+/// \param Input (optional) vtkMultiBlockDataSet Input trees
+/// \param Input (optional) vtkTable Info (such as clustering assigment)
+/// \param Output vtkMultiBlockDataSet Processed Input Trees
+/// \param Output vtkMultiBlockDataSet Origins
+/// \param Output vtkMultiBlockDataSet Geodesic Axes
+/// \param Output vtkMultiBlockDataSet Coefficients
+///
+/// This filter can be used as any other VTK filter (for instance, by using the
+/// sequence of calls SetInputData(), Update(), GetOutputDataObject()).
+///
+/// See the related ParaView example state files for usage examples within a
+/// VTK pipeline.
+///
+/// \b Related \b publication: \n
+/// "Wasserstein Auto-Encoders of Merge Trees (and Persistence Diagrams)" \n
+/// Mathieu Pont,  Julien Tierny.\n
+/// IEEE Transactions on Visualization and Computer Graphics, 2023
+///
+/// \sa ttk::MergeTreeAutoencoder
+/// \sa ttkAlgorithm
+
+#pragma once
+
+// VTK Module
+#include <ttkMergeTreeAutoencoderModule.h>
+
+// VTK Includes
+#include <ttkAlgorithm.h>
+#include <vtkMultiBlockDataSet.h>
+#include <vtkUnstructuredGrid.h>
+
+// TTK Base Includes
+#include <FTMTree.h>
+#include <MergeTreeAutoencoder.h>
+
+class TTKMERGETREEAUTOENCODER_EXPORT ttkMergeTreeAutoencoder
+  : public ttkAlgorithm // we inherit from the generic ttkAlgorithm class
+  ,
+    protected ttk::MergeTreeAutoencoder // and we inherit from the base class
+{
+private:
+  /**
+   * Add all filter parameters only as private member variables and
+   * initialize them here.
+   */
+  // Input options
+  double oldEpsilonTree1;
+
+  // ----------------------
+  // Data for visualization
+  // ----------------------
+  // Trees
+  std::vector<ttk::ftm::MergeTree<double>> intermediateDTrees;
+  std::vector<vtkUnstructuredGrid *> treesNodes, treesNodes2;
+  std::vector<vtkUnstructuredGrid *> treesArcs, treesArcs2;
+  std::vector<vtkDataSet *> treesSegmentation, treesSegmentation2;
+
+  void setDataVisualization(int ttkNotUsed(numInputs),
+                            int ttkNotUsed(numInputs2)) {
+  }
+
+  void resetDataVisualization() {
+    setDataVisualization(0, 0);
+    treesSegmentation.clear();
+    treesSegmentation2.clear();
+  }
+
+  // TODO
```

Comment:
what todo?
is the following function still needed?

Reply:
Removed in the next commit!

Reply:
Thanks!